### DISCUSSION TITLE
New design hiding the code by default is very inefficient

### DISCUSSION THREAD

**Discussion by korya** [community/community#53129] [Repositories]
### Select Topic Area

Product Feedback

### Body

A new UX design of Github repos where the code is hidden by default landed on me today. I assume the goal was to make the repo pages less cluttered. But I have to say it is very annoying. As a developer I need to access the code on my project more often then the documentation.

So hiding the code by default requires me to have an additional step to actually get to the real content. Very inconvenient and annoying on Desktop for sure.

I would love to move back to the Code+Readme layout. At least, to be able to opt-out from this new amazing feature. Is there a way?

**Comment by ZeroCool940711** [comment]
The new design is bad in general, on wide screens it's a pain to read anything compared to how things were more centered before and easier to read, also, the font looks like 10 times bigger than before or maybe its just because so much space is wasted that it feels like everything is bigger on the new layout but damn, it's a pain to read anything, I'm already looking for an extension or making one to revert everything to the old layout, I hate this kind of changes that improve nothing and just make things easier on mobile but a pain on desktop, I mean, it's a website for developers, I'm pretty sure most developers work on a desktop and check the repositories and code from a desktop and not a mobile phone 🤷‍♂️.

**Comment by davidleitw** [comment]
Agreed. Sometimes, the README contains information about the repository layout. Switching back and forth to view them is a terrible user experience. Basically, the README should be considered a part of the entire project layout to be complete. I can't think of any reason to separate the two. At the very least, please provide users with the option to return to the original mode. Thank you.

Updated: @ZeroCool940711  provided a way to turn off the new feature,so my previous statement that it couldn't be turned off was a misunderstanding. However, when turned off, the left sidebar with the file directory also disappears. I think these two features can be discussed separately. When developers are on the project homepage, the original README+Layout format should be maintained, and when they want to browse a specific file or folder, the left sidebar can provide a better experience. These two features should not be bundled together. Currently, if someone likes the left sidebar but dislikes the new design on the project homepage, they cannot turn off only one of these features.

**Reply by ZeroCool940711** [reply]
I found that you can disable it if you go to your profile picture on the top right, then click on `Feature Preview` and on `New Code Search and Code View` hit disable, believe me that I don't think I will ever enable that until the moment they force all of us to use that and then I will have to use a Chrome extension to revert the layout back as there is no way for me to use that new layout.

**Comment by firminochangani** [comment]
Honestly I feel quite frustrated due to the extra click I have to do in order to see the code. Gosh, just don't.

**Comment by fwilhe** [comment]
Please keep showing **code + readme content by default**. Having multiple readmes in multiple directories of a repo is a very common pattern for wiki style information. The toggle between code listing and readme content is a regression imo.
